x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -115,6 +116,8 @@ public class TransportDownsampleAction extends AcknowledgedTransportMasterNodeAc
     private final IndexScopedSettings indexScopedSettings;
     private final ThreadContext threadContext;
     private final PersistentTasksService persistentTasksService;
+    private final DownsampleMetrics downsampleMetrics;
+    private final long startTime;
```

Comment:
I think this is the wrong place. I think `startTime` should be a variable in the `masterOperation(...)` or we add `startTime` as a field to the `DownsampleAction.Request` class. Otherwise all downsample operations will have the same startTime and the duration will always increase.

Reply:
Oh my.. Done.